### PR TITLE
Add transfers commands and messages

### LINKED ISSUE #230: Implement transfer command
Add the transfer command, specs ongoing there: https://github.com/vegaprotocol/specs-internal/pull/779

### PR DESCRIPTION
**Pull request by jeremyletang**
close #230

### CODE REVIEW

File: commands/transfer_funds.go [hunk 1]
```diff
@@ -14,35 +20,77 @@ func checkTransfer(cmd *commandspb.Transfer) Errors {
...
+		errs.AddForProperty("transfer.to", ErrIsRequired)
+	}
+
+	if cmd.ToAccountType == vega.AccountType_ACCOUNT_TYPE_UNSPECIFIED {
+		errs.AddForProperty("transfer.to_account_type", ErrIsNotValid)
+	} else if _, ok := vega.AccountType_name[int32(cmd.ToAccountType)]; !ok {
+		errs.AddForProperty("transfer.to_account_type", ErrIsNotValid)
+	}
+
+	if cmd.FromAccountType == vega.AccountType_ACCOUNT_TYPE_UNSPECIFIED {
```
Comment (ze97286) [inline commands/transfer_funds.go:51]: not sure if it this is the place to verify but we also need to make sure we do not allow to transfer *from* one of the reward accounts.

Comment (jeremyletang) [reply commands/transfer_funds.go:51]: This is checked on the core.

Comment (ze97286) [reply commands/transfer_funds.go:51]: it's more a note to self to remember to check this :)

Comment (jeremyletang) [reply commands/transfer_funds.go:51]: ah fair enough

**Review by EVODelavega — APPROVED** [review]

**Pull request merged**